### ISSUE TITLE
Error raised by version 5.2.1+ when repo contains symlinks

### ISSUE THREAD

**Issue by andreipetre** [PyCQA/isort#1421]
I came across an issue [introduced by version 5.2.1](https://github.com/PyCQA/isort/commit/d16a72f750b8d523357b5535c62e95b603dfa794#diff-93a2daf0765c52e609aceb8df36da45eR125) where it would raise an error for a repo with symlinks to Python packages.

```bash
Traceback (most recent call last):
  File ".../bin/isort", line 8, in <module>
    sys.exit(main())
  File ".../isort/main.py", line 829, in main
    for sort_attempt in attempt_iterator:
  File ".../isort/main.py", line 818, in <genexpr>
    sort_imports(  # type: ignore
  File ".../isort/main.py", line 125, in iter_source_code
    dirnames.remove(dirname)
ValueError: list.remove(x): x not in list
```

`isort` 5.2.0 runs fine on the repo, 5.2.1 raises the above error. All versions after 5.2.0 (including 5.4.2) raise the same error.

**Comment by andreipetre** [comment]
This is already fixed on the `develop` branch (acd14d870978bec43b723e75e3cecb3e36837aaf).

**Comment by timothycrosley** [comment]
A fix for this issue has just been deployed to PyPI in version 5.5.0 of isort.

Thanks!

~Timothy
